### PR TITLE
Add mentoring notes for scala leap

### PR DESCRIPTION
**Pull request by ricemery**
Add mentoring notes for scala leap

### CODE REVIEW

File: tracks/scala/leap/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+### Reasonable solutions
+
+```scala
```
Comment (ErikSchierboom) [inline tracks/scala/leap/mentoring.md]: Just a suggestion. In the C# track, I use a sub-header to describe the various reasonable solutions if there is more than one (what makes them unique). For an example, see [the notes](https://github.com/exercism/website-copy/blob/master/tracks/csharp/exercises/bob/mentoring.md). You could consider adding those, but feel free to ignore.

File: tracks/scala/leap/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  def leapYear(year: Int): Boolean = {
+    (year % 4 == 0 && !(year % 100 ==0)) || year % 400 == 0
+  }
+}
+```
+
+
+### Common suggestions
+
+- True and false do not need to be returned explicitly. Instead return the result of combining expressions with &&, || and !
```
Comment (ErikSchierboom) [inline tracks/scala/leap/mentoring.md]:
```suggestion
- `true` and `false` do not need to be returned explicitly. Instead return the result of combining expressions with `&&`, `||` and `!`
```

File: tracks/scala/leap/mentoring.md (hunk 1 above)
Comment (ricemery) [reply tracks/scala/leap/mentoring.md]: Thanks!

**Comment by ErikSchierboom** [comment]
Nice!

**Pull request merged**